src/WebDav.Client/Request/SearchRequestBuilder.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace WebDav
+{
+    internal static class SearchRequestBuilder
+    {
+        /// <summary>
+        /// Build the WEBDAV XML body.
+        /// </summary>
+        /// <param name="select">Set arguments to select parameters</param>
+        /// <param name="fromDirectory">Set root directory to search from.</param>
+        /// <param name="searchKeyWord">Set search keyword, e.g. 'test%' % = like.</param>
+        /// <param name="where">Set argument where to search for.</param>
+        /// <param name="namespaces">Optional, add namespaces.</param>
```

Comment:
The `namespaces` parameter is nullable and when it is `null` then the method fails with `NullReferenceException` on `foreach (var ns in namespaces)` line.

My recommendation is to pass `SearchParameters` to this method and make sure collections are at least initialized with empty lists.

Reply:
Solved with initialization of all parameters.